### ISSUE TITLE
Doesn't work with typescript 5 moduleResolution set to bundler

### ISSUE THREAD

**Issue by Janpot** [developit/mitt#178]
To reproduce:

- install `typescript@beta` and `mitt`
- use following `tsconfig.json`:
    ```json
    {
      "compilerOptions": {
        "lib": ["esnext"],
        "strict": true,
        "moduleResolution": "bundler"
      }
    }
    ```
- import `mitt`:
    ```ts
    import mitt from "mitt";

    const emitter = mitt();
    ```
- run `npx tsc`
- observe error:
    ```
        index.ts:1:18 - error TS7016: Could not find a declaration file for module 'mitt'. '/mittrepro/node_modules/mitt/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'mitt' library may need to update its package.json or typings.

    1 import mitt from "mitt";
                      ~~~~~~


    Found 1 error in index.ts:1
    ```

Can be fixed by changing package.json exports to:

```json
  "exports": {
    "types": "./index.d.ts",
    "import": "./dist/mitt.mjs",
    "require": "./dist/mitt.js",
    "default": "./dist/mitt.mjs"
  },
```

**Comment by Janpot** [comment]
Looks like https://github.com/developit/mitt/commit/1194efba1adb575a1fdb26d7638f448faf5df2d4 fixes it, but it never got published

**Comment by CxC-j** [comment]
thanks

**Comment by DamianGlowala** [comment]
@developit

**Comment by developit** [comment]
I will try to get this fixed and published in the next couple of days.

**Comment by phiter** [comment]
> I will try to get this published in the next couple of days.

Thanks for the response @developit. Looking forward to updating!

**Comment by ajn404** [comment]
if your mitt is "^3.0.0 "
in **tsconfig.json** add 
```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "mitt": ["./node_modules/mitt/index.d.ts"]
    }
  }
}
```
may help

**Comment by ajn404** [comment]
@phiter sorry,my judgment may not be accurate.But this method of configuring tsconfig is valid in [my project ](https://github.com/ajn404/new_pad/blob/main/tsconfig.json).I guess😊 in your config ,maybe path is  not  correct,so that ts complier can not find the index.d.ts for mitt

**Comment by sampotts** [comment]
@developit Can you publish the latest version? That'd be awesome.

**Comment by developit** [comment]
Sorry for the super long wait folks! This fix has been published as [mitt@3.0.1](https://github.com/developit/mitt/releases/tag/3.0.1).
